staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -297,11 +297,11 @@ func (j *JSONPath) findFieldInValue(value *reflect.Value, node *FieldNode) (refl
 	var inlineValue *reflect.Value
 	for ix := 0; ix < t.NumField(); ix++ {
 		f := t.Field(ix)
-		jsonTag := f.Tag.Get("json")
-		parts := strings.Split(jsonTag, ",")
-		if len(parts) == 0 {
+		jsonTag, found := f.Tag.Lookup("json")
```

Comment:
WRT running jsonpath on the serialization - I'm actually not sure whether there's a 1:1 mapping between objects and their serializations, e.g. are these equivalent serializations of `SomeType{ObjectMeta:{Name: "foo"}}`, because ObjectMeta is inlined?
```
{"objectMeta":{"name":"foo"}}
```
```
{"name":"foo"}
```
If they are equal, wouldn't people expect the filters `{.name}` and `{.objectMeta.name}` to work the same? I think the only way to get around this is to run jsonpath on the deserialized object, which knows what is inline.

It would probably be good to share code between the decoder and jsonpath. This is one of those situations where I really hope we don't need bug-for-bug compatibility with old clients.